applications/poisson/sine/application.h

```diff
@@ -286,8 +279,11 @@ class Application : public ApplicationBase<dim, n_components, Number>
     pp_data.output_data.write_output       = this->output_parameters.write;
     pp_data.output_data.directory          = this->output_parameters.directory + "vtu/";
     pp_data.output_data.filename           = this->output_parameters.filename;
-    pp_data.output_data.write_higher_order = true;
-    pp_data.output_data.degree             = this->param.degree;
+    pp_data.output_data.write_higher_order = false;
+    if(pp_data.output_data.write_higher_order == true and dim == 3 and
+       this->param.grid.element_type == ElementType::Simplex)
+      AssertThrow(false, dealii::ExcMessage("Can't use higher order output with 3D Simplices."));
```

Comment:
I think this can be written more elegantly:

```c++
    if(this->param.grid.element_type == ElementType::Simplex and dim == 3)
      AssertThrow(pp_data.output_data.write_higher_order == false, dealii::ExcMessage("Can't use higher order output with 3D Simplices."));
```

Reply:
The following is also an option:

```suggestion
    if(dim == 3 and this->param.grid.element_type == ElementType::Simplex)
       pp_data.output_data.write_higher_order = false;
    else
       pp_data.output_data.write_higher_order = true;
```

I like this more, because it makes switching between element types easier.

Reply:
sound good as well, but we still need the assert in the code, at best at a central place so that we only have to write this once. The (minor) problem with your suggestion is that from the code one cannot see that the combination Simplex with `write_higher_order` does not work. One might guess this, but it is not entirely clear (and also not enforced).